src/HeaderCell.tsx

```diff
@@ -180,13 +195,18 @@ export default function HeaderCell<R, SR>({
   }
 
   function onDragStart(event: React.DragEvent<HTMLDivElement>) {
-    event.dataTransfer.setData(dragDropKey, column.key);
+    const dragImage = event.currentTarget.cloneNode(true) as HTMLDivElement;
+    dragImage.classList.add(dragImageClassname);
+    dragImage.id = dragImageId;
+    event.currentTarget.parentElement!.insertBefore(dragImage, event.currentTarget);
+    event.dataTransfer.setDragImage(dragImage, 0, 0);
     event.dataTransfer.dropEffect = 'move';
-    setIsDragging(true);
+    setDraggedColumnKey(column.key);
   }
 
   function onDragEnd() {
-    setIsDragging(false);
+    setDraggedColumnKey(undefined);
+    document.getElementById(dragImageId)!.remove();
```

Comment:
What if React removes it during a re-render?
```suggestion
    document.getElementById(dragImageId)?.remove();
```

Reply:
I don't think it should happen but cannot say for sure. I will check if there is a way to render another element instead of cloning